store/cassandra/cassandra.go

```diff
@@ -355,6 +355,21 @@ func (c *CassandraStore) insertChunk(key string, t0, ttl uint32, data []byte) er
 		return nil
 	}
 
+	// we calculate ttl like this:
+	// - the chunk's t0 plus maxChunkSpan is the ts of last possible datapoint in the chunk
+	// - the timestamp of the last datapoint + ttl is the timestamp until when we want to keep this chunk
+	// - then we subtract the current time stamp to get the difference relative to now
+	// - the result is the ttl in seconds relative to now
+	relativeTtl := int64(t0+mdata.MaxChunkSpan()+ttl) - time.Now().Unix()
```

Comment:
My concern with using MaxChunkSpan is that it makes the minimum ttl of data at MaxChunkSpan.  eg, if you want raw data stored for 1h, but also have rollups being stored for longer and those rollups use a chunkSpan of 6h, then the raw data will be stored for a lot longer then intended.

This is not a concern for any of our use cases, but might cause problems for other users and perhaps end2end tests.

Reply:
That's a valid concern, but I'm not sure how to improve that. There are basically two possibilities:

1) we read the data byte slice to determine the chunk span. this should only require the first 2 bytes, it can be done like here: https://github.com/grafana/metrictank/blob/master/mdata/chunk/itergen.go#L76
2) we add a span property to the chunk write request's payload. but this will require updating all locations that generate chunk write requests.

I'm leaning towards 1)

Reply:
I've given `1)` a try: https://github.com/grafana/metrictank/pull/1448/commits/a4a967556c2045ade3cfbec50b7395920316ec43

Reply:
I ran another test with the latest version. I've started MT with an empty cassandra instance and this schema:
```
1s:1h:10min:1,1min:48h:6h:1
```
Then I fed it with 72h of data for one metric. It created two tables for the two different TTLs `metric_1`/`metric_32`. All the TTLs in these two tables look like what we want:

```
cqlsh:metrictank> select key, ts, TTL(data) from metric_1;

 key                                    | ts         | ttl(data)
----------------------------------------+------------+-----------
 1.d0a8110e69b0d874610aa08ab6740dfa_647 | 1567179600 |      3815
 1.d0a8110e69b0d874610aa08ab6740dfa_647 | 1567179000 |      3215
 1.d0a8110e69b0d874610aa08ab6740dfa_647 | 1567178400 |      2615
 1.d0a8110e69b0d874610aa08ab6740dfa_647 | 1567177800 |      2015
 1.d0a8110e69b0d874610aa08ab6740dfa_647 | 1567177200 |      1415
 1.d0a8110e69b0d874610aa08ab6740dfa_647 | 1567176600 |       815
 1.d0a8110e69b0d874610aa08ab6740dfa_647 | 1567176000 |       215

(7 rows)
cqlsh:metrictank> select key, ts, TTL(data) from metric_32;

 key                                           | ts         | ttl(data)
-----------------------------------------------+------------+-----------
 1.d0a8110e69b0d874610aa08ab6740dfa_sum_60_647 | 1567144800 |    159128
 1.d0a8110e69b0d874610aa08ab6740dfa_sum_60_647 | 1567123200 |    137528
 1.d0a8110e69b0d874610aa08ab6740dfa_sum_60_647 | 1567101600 |    115928
 1.d0a8110e69b0d874610aa08ab6740dfa_sum_60_647 | 1567080000 |     94328
 1.d0a8110e69b0d874610aa08ab6740dfa_sum_60_647 | 1567058400 |     72728
 1.d0a8110e69b0d874610aa08ab6740dfa_sum_60_647 | 1567036800 |     51128
 1.d0a8110e69b0d874610aa08ab6740dfa_sum_60_647 | 1567015200 |     29528
 1.d0a8110e69b0d874610aa08ab6740dfa_sum_60_647 | 1566993600 |      7928
 1.d0a8110e69b0d874610aa08ab6740dfa_min_60_647 | 1567144800 |    159128
 1.d0a8110e69b0d874610aa08ab6740dfa_min_60_647 | 1567123200 |    137528
 1.d0a8110e69b0d874610aa08ab6740dfa_min_60_647 | 1567101600 |    115928
 1.d0a8110e69b0d874610aa08ab6740dfa_min_60_647 | 1567080000 |     94328
 1.d0a8110e69b0d874610aa08ab6740dfa_min_60_647 | 1567058400 |     72728
 1.d0a8110e69b0d874610aa08ab6740dfa_min_60_647 | 1567036800 |     51128
 1.d0a8110e69b0d874610aa08ab6740dfa_min_60_647 | 1567015200 |     29528
 1.d0a8110e69b0d874610aa08ab6740dfa_min_60_647 | 1566993600 |      7928
 1.d0a8110e69b0d874610aa08ab6740dfa_cnt_60_647 | 1567144800 |    159128
 1.d0a8110e69b0d874610aa08ab6740dfa_cnt_60_647 | 1567123200 |    137528
 1.d0a8110e69b0d874610aa08ab6740dfa_cnt_60_647 | 1567101600 |    115928
 1.d0a8110e69b0d874610aa08ab6740dfa_cnt_60_647 | 1567080000 |     94328
 1.d0a8110e69b0d874610aa08ab6740dfa_cnt_60_647 | 1567058400 |     72728
 1.d0a8110e69b0d874610aa08ab6740dfa_cnt_60_647 | 1567036800 |     51128
 1.d0a8110e69b0d874610aa08ab6740dfa_cnt_60_647 | 1567015200 |     29528
 1.d0a8110e69b0d874610aa08ab6740dfa_cnt_60_647 | 1566993600 |      7928
 1.d0a8110e69b0d874610aa08ab6740dfa_max_60_647 | 1567144800 |    159128
 1.d0a8110e69b0d874610aa08ab6740dfa_max_60_647 | 1567123200 |    137528
 1.d0a8110e69b0d874610aa08ab6740dfa_max_60_647 | 1567101600 |    115928
 1.d0a8110e69b0d874610aa08ab6740dfa_max_60_647 | 1567080000 |     94328
 1.d0a8110e69b0d874610aa08ab6740dfa_max_60_647 | 1567058400 |     72728
 1.d0a8110e69b0d874610aa08ab6740dfa_max_60_647 | 1567036800 |     51128
 1.d0a8110e69b0d874610aa08ab6740dfa_max_60_647 | 1567015200 |     29528
 1.d0a8110e69b0d874610aa08ab6740dfa_max_60_647 | 1566993600 |      7928

(32 rows)
```